shell/key-bindings.zsh

```diff
@@ -112,7 +112,7 @@ fzf-history-widget() {
   # history lines, is loaded, and that Perl is installed for multi-line output.
   if zmodload -F zsh/parameter p:history 2>/dev/null && (( ${#commands[perl]} )); then
     selected="$(printf '%1$s\t%2$s\000' "${(vk)history[@]}" |
```

Comment:
```suggestion
    selected="$(printf '%s\t%s\000' "${(@kv)history}" |
```

The numbers in the format part of the `printf` command were added by me, but they are superfluous.

```sh
man printf
…
n$          Allows reordering of the output according to argument.
```

As for switching `vk` to `kv`, it does not affect the outcome, but it provides a more logical order for the reader, where 'key' precedes 'value'. And moving the `@` inside the `()` allows us to remove the `[]` saving two characters.

```sh
man zshexpn
…
       Parameter  expansions  can  involve flags, as in `${(@kv)aliases}'
…
       @      In double quotes, array elements are put  into  separate  words.
              E.g.,   `"${(@)foo}"'   is   equivalent   to  `"${foo[@]}"'  and
              `"${(@)foo[1,2]}"' is the same as `"$foo[1]"  "$foo[2]"'.   This
              is  distinct  from field splitting by the f, s or z flags, which
              still applies within each array element.
```

Reply:
Thanks for the explanation.

Reply:
One question, though. Is this supported by the old versions of zsh?

Reply:
I checked with `5.0.2`.

```sh
❯ ~/.local/zsh-test/bin/zsh-5.0.2 -f
Paria% echo test
test
Paria% seq 2
1
2
Paria% printf '%s\t%s\n' "${(@kv)history}"
2	seq 2
1	echo test
Paria% 
```

Reply:
But checking the source code, this seems to be a very old feature, reaching back into the nineties.

Reply:
Thanks!